bigquery_etl/docs/generate_docs.py

```diff
@@ -86,14 +87,31 @@ def main():
                         dest = project_doc_dir / "overview.md"
                         dest.write_text(load_with_examples(src))
                     else:
+                        description = None
+                        if METADATA_FILE in files:
+                            with open(os.path.join(root, METADATA_FILE)) as stream:
+                                try:
+                                    description = yaml.safe_load(stream).get(
+                                        "description", None
+                                    )
+                                except yaml.YAMLError:
+                                    pass
                         # dataset or UDF level doc file
                         if UDF_FILE in files:
                             # UDF-level doc; append to dataset doc
                             dataset_name = os.path.basename(path)
                             dataset_doc = out_dir / path.parent / f"{dataset_name}.md"
+                            docfile_content = load_with_examples(src)
                             with open(dataset_doc, "a") as dataset_doc_file:
                                 dataset_doc_file.write("\n\n")
-                                dataset_doc_file.write(load_with_examples(src))
+                                # Inject a level-2 header with the UDF name
+                                if not docfile_content.strip().startswith("#"):
+                                    dataset_doc_file.write(f"## {name}\n\n")
+                                # Inject the "description" from metadata.yaml
+                                if description:
+                                    dataset_doc_file.write(f"{description}\n\n")
+                                # Inject the contents of the README.md
+                                dataset_doc_file.write(docfile_content)
```

Comment:
If the UDF-specific doc file does have a section starting with `#`, we'd prepend the description which would be followed by the section. So, I think we could end up with something like:

```
Some description - foo bar

## get_key (or some section name)

content of the README.md
```

Reply:
I'll remove this conditional. I think we should always expect for the function name to be injected.